components/pedigree/resources/src/main/resources/pedigree/model/phenotipsJSON.js

```diff
@@ -0,0 +1,260 @@
+            }
+            if (patientJSON.patient_name.hasOwnProperty("last_name")) {
+                result.lName = patientJSON.patient_name.last_name;
+            } else {
+                result.lName = "";
+            }
+        }
+
+        if (patientJSON.hasOwnProperty("sex")) {
+            result.gender = patientJSON.sex;
+        } else {
+            result.gender = "U";
+        }
+
+        if (patientJSON.hasOwnProperty("date_of_birth")) {
+            var birthDate = new PedigreeDate(patientJSON.date_of_birth);
+            result.dob = birthDate.getSimpleObject();
+        } else {
+            delete result.dob;
+        }
+        if (patientJSON.hasOwnProperty("date_of_death")) {
+            var deathDate = new PedigreeDate(patientJSON.date_of_death);
+            result.dod = deathDate.getSimpleObject();
+            if (deathDate.isSet()) {
+                delete result.aliveandwell;
+            }
+        } else {
+            delete result.dod;
+        }
+        if (patientJSON.hasOwnProperty("life_status")) {
+            var lifeStatus = patientJSON["life_status"];
+            if (lifeStatus == "deceased" || lifeStatus == "alive") {
+                result.lifeStatus = lifeStatus;
+            }
+            if (lifeStatus != "alive") {
+                // if not removed, it will overwrite the life status to 'alive' and thus remove death date
+                delete result.aliveandwell;
+            }
+        } else {
+            delete result.lifeStatus;
+        }
+
+        if (patientJSON.hasOwnProperty("ethnicity")) {
+            // e.g.: "ethnicity":{"maternal_ethnicity":["Yugur"],"paternal_ethnicity":[]}
+            var ethnicities = [];
+            if (patientJSON.ethnicity.hasOwnProperty("maternal_ethnicity")) {
+                ethnicities = patientJSON.ethnicity.maternal_ethnicity.slice(0);
+            }
+            if (patientJSON.ethnicity.hasOwnProperty("paternal_ethnicity")) {
+                ethnicities = ethnicities.concat(patientJSON.ethnicity.paternal_ethnicity.slice(0));
+            }
+            if (ethnicities.length > 0) {
+                result.ethnicities = Helpers.filterUnique(ethnicities);
+            }
+        }
+
+        if (patientJSON.hasOwnProperty("external_id")) {
+            result.externalID = patientJSON.external_id;
+        } else {
+            delete result.externalID;
+        }
+
+        if (patientJSON.hasOwnProperty("features") && patientJSON.features.length > 0) {
+            result.features = patientJSON.features;
+        } else {
+            delete result.features;
+        }
+
+        if (patientJSON.hasOwnProperty("nonstandard_features") && patientJSON.nonstandard_features.length > 0) {
+            result.nonstandard_features = patientJSON.nonstandard_features;
+        } else {
+            delete result.nonstandard_features;
+        }
+
+        var disorders = [];
+        if (patientJSON.hasOwnProperty("disorders")) {
```

Comment:
offtopic: Why don't we support `clinical_diagnosis` aka Orphanet diseases?

Reply:
just because no one implemented the support :)